### PR TITLE
created new branch with tasks for new PR

### PR DESCRIPTION
**Pull request by lensman75**
New branch with previous done tasks for correct PR.

### CODE REVIEW

**Comment by lensman75** [comment]
Can't finish logicalCalculator.js
Can't path through tests.

File: carYearsDogYears.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      dogYear = 15 + 9;
+      break;
+    default:
+      catYear = 15 + 9;
+      dogYear = 15 + 9;
+      break;
+      return catYear, dogYear;
+  }
+  if (humanYears > 2) {
+    for (let i = 2; i < humanYears; i++) {
```
Comment (mkl-) [inline carYearsDogYears.js:23]: You do not need loop here.
`catYears += otherCatyears * (humanYears - 2)`

File: carYearsDogYears.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+      break;
+    case 2:
+      catYear = 15 + 9;
+      dogYear = 15 + 9;
+      break;
+    default:
+      catYear = 15 + 9;
+      dogYear = 15 + 9;
+      break;
+      return catYear, dogYear;
```
Comment (mkl-) [inline carYearsDogYears.js:20]: This `return` is not needed.

File: countingSheeps.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
+// https://www.codewars.com/kata/54edbc7200b811e956000556/train/javascript
+function countSheeps(arrayOfSheep) {
+  let count = 0;
+  for (let i = 0; i < arrayOfSheep.length; i++) {
+    if (arrayOfSheep[i] !== true) {
```
Comment (mkl-) [inline countingSheeps.js:5]: I guess simple `if` is sufficient
```
if(arrayOfSheep[i] === true) {
   count += 1
}
```

File: logicalCalculator.js [hunk 4]
```diff
@@ -1,43 +1,52 @@
...
-//     case 'AND':
-//       if (a1 == true && a2 == true) {
-//         return true;
-//       } else return false;
-//   }
-// }
-
 function logicalCalc(array, op) {
-  if (array.length ==1) {
+  if (op === 'XOR' && array.length == 1) {
```
Comment (mkl-) [inline logicalCalculator.js:3]: You do not need this check.

File: logicalCalculator.js [hunk 5]
```diff
@@ -1,43 +1,52 @@
...
+  if (op === 'OR') {
+    let res = array.reduce(function (one, two) {
+      return one || two;
+    }, array[0]);
+    return res;
+  }
+  if (op === 'XOR') {
+    let res = array.reduce(function (one, two) {
+      return one ^ two;
+    }, array[0]);
```
Comment (mkl-) [inline logicalCalculator.js:21]: It is incorrect.
You cannot use array[0] as a starting value for XOR operation.
Test:  
` assert.strictEqual(logicalCalc([false, true], "XOR"), true); `

Comment (mkl-) [reply logicalCalculator.js:21]: It is better to use something like
```
array.reduce(function (one, two) {
      return one ^ two;
    });
```

File: logicalCalculator.js [hunk 6]
```diff
@@ -1,43 +1,52 @@
...
+    if (res == 1) {
+      return false;
```
Comment (mkl-) [inline logicalCalculator.js:23]: I guess here should be `true`.

File: removeExclamationMark.js [hunk 7]
```diff
@@ -0,0 +1,14 @@
+// https://www.codewars.com/kata/57a0885cbb9944e24c00008e/train/javascript
+
+function removeExclamationMarks(s) {
+  let arr = s.split('');
+  let newArr = [];
+  for (let i = 0; i < arr.length; i++) {
+    if (arr[i] == '!') {
```
Comment (mkl-) [inline removeExclamationMark.js:7]: I guess
```
if(arr[i]  !== '!') {
   newArr.push(arr[i]);
}
```
would be simpler

**Review by mkl- — CHANGES_REQUESTED** [review]

**Pull request merged**